module/VuFind/src/VuFind/Search/ProQuestFSG/Results.php

```diff
@@ -0,0 +1,175 @@
+
+use VuFindSearch\Command\SearchCommand;
+
+use function strlen;
+
+/**
+ * ProQuest Federated Search Gateway Search Parameters
+ *
+ * @category VuFind
+ * @package  Search_ProQuestFSG
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class Results extends \VuFind\Search\Base\Results
+{
+    /**
+     * Search backend identifier.
+     *
+     * @var string
+     */
+    protected $backendId = 'ProQuestFSG';
+
+    /**
+     * Facets returned in search response.
+     *
+     * @var array
+     */
+    protected $responseFacets = null;
+
+    /**
+     * Simplified version of result facets.
+     *
+     * @var array
+     */
+    protected $simplifiedResponseFacets = null;
+
+    /**
+     * Support method for performAndProcessSearch -- perform a search based on the
+     * parameters passed to the object.
+     *
+     * @return void
+     */
+    protected function performSearch()
+    {
+        $query  = $this->getParams()->getQuery();
+        $allTerms = $query->getAllTerms();
+        if (!strlen($allTerms)) {
+            $this->storeErrorResponse('empty_search_disallowed');
+            return;
+        }
+        $limit  = $this->getParams()->getLimit();
+        $offset = $this->getStartRecord();
+        $params = $this->getParams()->getBackendParameters();
+        $command = new SearchCommand(
+            $this->backendId,
+            $query,
+            $offset,
+            $limit,
+            $params
+        );
+        $collection = $this->getSearchService()
+            ->invoke($command)->getResult();
+
+        $this->resultTotal = $collection->getTotal();
+        $this->results = $collection->getRecords();
+
+        // ProQuest does not return facets unless the startRecord is 1
+        if ($offset === 1) {
+            $this->responseFacets = $collection->getFacets();
+            $this->simplifiedResponseFacets = $this->simplifyFacets($this->responseFacets);
+        }
+    }
+
+    /**
+     * Store an empty response with an error message instead of performing a search.
+     *
+     * @param string|array $error Error message(s) to display to user.
+     *
+     * @return void
+     */
+    protected function storeErrorResponse(string|array $error): void
+    {
+        $this->resultTotal = 0;
+        $this->results = [];
+        $this->simplifiedResponseFacets = [];
+        $this->responseFacets = [];
+        $this->errors = (array)$error;
+    }
+
+    /**
+     * Returns the stored list of facets for the last search
+     *
+     * @param array $filter Array of field => on-screen description listing
+     * all of the desired facet fields; set to null to get all configured values.
+     *
+     * @return array        Facets data arrays
+     */
+    public function getFacetList($filter = null)
+    {
+        if (null === $this->simplifiedResponseFacets) {
```

Comment:
Should we also check here whether any facets are actually turned on? You should be able to check `$this->getParams()->getFacetConfig()` and bypass the extra lookup if the Database facet has not been requested.

Reply:
This is a good point on efficiency in case someone turns off the Database facet, but I think I can just check $filter passed into this method to see if it's empty?

Reply:
A null $filter value means "get all configured facets," so you might need to check the config if you get a null value... but you're right that if it's a non-null value, you can look at the filter and not worry about the config.

Reply:
I implemented the $filter check and it seems to work correctly.

Reply:
Ah sorry posted my comment before seeing your last one...will review.

Reply:
I think you might want to add a `&& $filter !== null` check in there to be safe; I can't remember under what circumstances the null value is passed in, but if you don't check for null, then the facets will be excluded when somebody asks for all the facets, which seems likely to cause a problem somewhere.

Reply:
You were right the first time around (I think) -- I swapped out the $filter logic with `getFacetConfig` and it seems to work as expected in all cases, including if I hard code `$filter = null` they still display.